modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/VolatileTxStateMetaStorage.java

```diff
@@ -155,14 +187,62 @@ public void vacuum(long vacuumObservationTimestamp, long txnResourceTtl) {
             });
         });
 
-        LOG.info("Vacuum finished [vacuumObservationTimestamp={}, txnResourceTtl={}, vacuumizedTxnsCount={},"
-                + " markedAsInitiallyDetectedTxnsCount={}, alreadyMarkedTxnsCount={}, skippedFotFurtherProcessingUnfinishedTxnsCount={}].",
-                vacuumObservationTimestamp,
-                txnResourceTtl,
-                vacuumizedTxnsCount,
-                markedAsInitiallyDetectedTxnsCount,
-                alreadyMarkedTxnsCount,
-                skippedFotFurtherProcessingUnfinishedTxnsCount
+        beforeVacuum.apply(txIds)
+                .thenAccept(tuple -> {
```

Comment:
Seems that we may even skip the future, and just update volatile txn state after successful persistent txn state removal by setting cleanupCompletionTimestamp to null if it's the same as was propagated to the function(currently there's no such propagation). In that case next vacuum iteration will just remove the state from the volatile map. What do you think?

Reply:
kinda tricky, and we will have to pass the timestamps map to the persistent vacuum operation in order to compare the timestamps. Also, we'd better to have to count of successful persistent vacuums in log, so we cant just skip this future. 

But I slightly revorked this part.